app/services/PageBuilder.scala

```diff
@@ -133,6 +133,8 @@ class PageBuilder extends ProcessPopulation {
           f(page.id, page.url, text.langs(0))
```

Comment:
The methods fromPageDetails and pagesWithValidation are used only in the test classes which makes me wonder if they should be relocated into the test classes.

Reply:
Richard, fromPageDetails and pagesWithValidation are both used in EG and this code is shared between both the viewer and EG

Reply:
Ok, I can one usage in external-guidance.